controllers/mirrorpeer_controller.go

```diff
@@ -602,38 +602,22 @@ func (r *MirrorPeerReconciler) processManagedClusterAddon(ctx context.Context, m
 	for _, config := range addonConfigs {
 		logger.Info("Handling ManagedClusterAddon for cluster", "ClusterName", config.Namespace)
 
-		var managedClusterAddOn addonapiv1alpha1.ManagedClusterAddOn
-		namespacedName := types.NamespacedName{
-			Name:      config.Name,
-			Namespace: config.Namespace,
-		}
-
-		err := r.Client.Get(ctx, namespacedName, &managedClusterAddOn)
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				logger.Info("ManagedClusterAddon not found, will create a new one", "ClusterName", config.Namespace)
-
-				annotations := make(map[string]string)
-				annotations[utils.DRModeAnnotationKey] = string(mirrorPeer.Spec.Type)
-				annotations[AddonVersionAnnotationKey] = version.Version
-				managedClusterAddOn = addonapiv1alpha1.ManagedClusterAddOn{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:        setup.TokenExchangeName,
-						Namespace:   config.Namespace,
-						Annotations: annotations,
-					},
-				}
-			}
+		managedClusterAddOn := addonapiv1alpha1.ManagedClusterAddOn{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      setup.TokenExchangeName,
+				Namespace: config.Namespace,
+			},
 		}
 
 		logger.Info("Installing agents on the namespace", "InstallNamespace", config.InstallNamespace)
 		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, &managedClusterAddOn, func() error {
+			annotations := make(map[string]string)
+			annotations[utils.DRModeAnnotationKey] = string(mirrorPeer.Spec.Type)
```

Comment:
Is this spec required to be sent ? Since we are just creating s3 buckets in async or sync case

Reply:
Probably not required but I don't want to change it as that is a separate concern. Eventually I plan to enable addons to handle both modes simultaneously. This should be resolved at that point.

Reply:
ok, can manage this for separate PR